lib/elixir/lib/uri.ex

```diff
@@ -977,6 +977,35 @@ defmodule URI do
       %{uri | query: uri.query <> "&" <> query}
     end
   end
+
+  @doc """
+  Joins `path` to the given `uri`.
+
+  Any additional url components in `path` (like query strings) will be removed.
```

Comment:
Tesla, Req, and probably other HTTP clients have a base_url feature. In Req we can do stuff like: `Req.new(base_url: "https://example.com") |> Req.get("/foo")`.

I'd love to use a standard library for merging the urls but dropping the query string would be a deal breaker for me.

I guess I could implement it like this, would that be the way to go do you think?

```elixir
def merge(base_url, url) when url.host != nil do
  url
end

def merge(base_url, url) do
  base_url
  |> URI.append_path(url.path)
  |> URI.append_query(url.query)
end
```

Reply:
We should definitely not drop the query string nor the anchor. The user can do that if desired.

Reply:
If we don't drop them, do we automatically merge them? I.e.: `append_path("/foo?x=1", "/bar?y=2") #=> "/foo/bar?x=1&y=2"`?

Reply:
I would not expect path to have a query string. Then it would be innaccurately named or we need a whole different function altogether.